src/v/model/metadata.h

```diff
@@ -196,6 +196,10 @@ class broker {
 /// type representing single replica assignment it contains the id of a broker
 /// and id of this broker shard.
 struct broker_shard {
+    using value_t = broker_shard;
+    static constexpr serde::version_t redpanda_serde_version = 0;
```

Comment:
We should add explicit adl serialization so that we can inherit from envelope and start off from a clean slate. This has the added benefit of forcing devs to update both adl and serde in parallel as we will have to support both for some period of time.

Reply:
I started doing this for the RPC types. Seems to be pretty straight forward. I guess the alternative approach is to wait until we fully remove ADL support and then add back the serde::envelope inheritance?